sql/src/main/java/org/apache/druid/sql/AbstractStatement.java

```diff
@@ -49,6 +57,14 @@ public abstract class AbstractStatement implements Closeable
   protected final SqlToolbox sqlToolbox;
   protected final SqlQueryPlus queryPlus;
   protected final SqlExecutionReporter reporter;
+
+  /**
+   * Copy of the query context modified during planning. Modifications are
+   * valid in tasks that run in the request thread. Once the query forks
+   * child threads, then concurrent modifications to the query context will
+   * result in an undefined muddle of race conditions.
+   */
+  protected final Map<String, Object> queryContext;
```

Comment:
So this is mutable?

Reply:
Yes. Revised the comment to make the semantics a bit clearer. See if the revised text helps.